pandas/core/array_algos/take.py

```diff
@@ -142,8 +148,11 @@ def take_1d(
         # ExtensionArray -> dispatch to their method
         return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill)
 
+    if not allow_fill:
```

Comment:
good!  i was gonna mention i got segfaults when i tried using this elsewhere in this case